crates/module-system/module-implementations/sov-evm/src/rpc/mod.rs

```diff
@@ -334,6 +346,9 @@ where
                 match pending_or_block_nr {
                     PendingOrBlock::Pending => MaybeArchivalState::Current(state),
                     PendingOrBlock::Number(number) => {
+                        if number == state.rollup_height_to_access().get() || (number == state.rollup_height_to_access().get() + 1) {
```

Comment:
This is just a minor nit, and the PR can be merged without addressing it. Inside `str_to_block_nr`, we detect the `latest` block number using `self.block_numbers(state).end()`, so we could reuse the same logic here for consistency. Or `str_to_block_nr` could even return an enum `PendingOrLatestBlock` and we just pattern match on it.